lib/asserts.nix

```
@@ -2,78 +2,122 @@
 
 rec {
 
-  /* Throw if pred is false, else return pred.
-     Intended to be used to augment asserts with helpful error messages.
+  /**
+    Throw if pred is false, else return pred.
+    Intended to be used to augment asserts with helpful error messages.
 
-     Example:
-       assertMsg false "nope"
-       stderr> error: nope
+    # Example
 
-       assert assertMsg ("foo" == "bar") "foo is not bar, silly"; ""
-       stderr> error: foo is not bar, silly
+    ```nix
+    assertMsg false "nope"
+    stderr> error: nope
+    assert assertMsg ("foo" == "bar") "foo is not bar, silly"; ""
+    stderr> error: foo is not bar, silly
```

This is a separate comment because I'm ok making it optional and doing a pass on all of this later, since it involves a lot more work.
(applies to other functions too)

The [conventions](https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#documentation-conventions) we reached make it easier for people to understand that this is code inside a repl and for them to replicate the example:

```suggestion
    nix-repl> pkgs.lib.assertMsg false "nope"
    error:
           … in the right operand of the OR (||) operator
           (some output removed for clarity)
           error: nope
    nix-repl> assert pkgs.lib.assertMsg ("foo" == "bar") "foo is not bar, silly"; ""
    error:
           … in the condition of the assert statement
           (some output removed for clarity)
           error: foo is not bar, silly
```

---

While this looks much better we unfortunately cannot do this automatically with the codemod. 
fyi: You can try it yourself
`nix run github:nix-community/noogle#codemod <filename or folder>`
This is in fact how #262987 is created.

My goal is to batch out small merges to reduce the changeset of this big PR to something close to 0

---

Yeah, I'm aware this isn't possible to be done automatically and will need someone to manually adjust all the code, which is why I mentioned it's optional and we can do a pass to improve the contents later.